### ISSUE TITLE
Visualized BGE Evaluation

### ISSUE THREAD

**Issue by zwhus** [FlagOpen/FlagEmbedding#948]
Hi, i want to reproduce the result of Visualized BGE, but zero-shot benchmark not clear, such as WebQA. Can you provide evaluation dataset and codes for zero-shot  benchmark. Thanks!

**Comment by JUNJIE99** [comment]
Sure, I will release the relevant evaluation datasets and evaluation code soon. I will inform you when it is complete. Thank you for your attention and patience.

**Comment by zwhus** [comment]
Thanks, can you provide a time for me?

**Comment by JUNJIE99** [comment]
Hello, the WebQA dataset and evaluation code have been made available at [here](https://github.com/JUNJIE99/VISTA_Evaluation). Should you have any further questions, please feel free to reach out.

We will also be progressively uploading other evaluation datasets.

Thanks.

**Comment by zwhus** [comment]
Thanks

**Comment by zwhus** [comment]
Thanks, Can you provide other benchmark such as FashIQ and CIRR? I notice CIRR score is 23.9(R1) in Pixword, but in paper is 23.42(R5)

**Comment by zwhus** [comment]
Thank you for your quick response. I have noticed this detail. However, there is no CIRR image in the link. After I installed Pix2Word and downloaded it, there are 16,939 training images and 2,315 for testing and validation, with a total of 21,569 images. It seems there is a slight difference. Is there any additional operation required?

**Comment by JUNJIE99** [comment]
I apologize, it seems that the CIRR image upload was interrupted earlier due to network issues, and I am currently re-uploading them. If all goes well, the upload should be successful within half an hour. If you still haven't seen it by the end of today, please feel free to let me know.

Regarding the slight difference, I checked the CIRR dataset [paper](https://openaccess.thecvf.com/content/ICCV2021/papers/Liu_Image_Retrieval_on_Real-Life_Images_With_Pre-Trained_Vision-and-Language_Models_ICCV_2021_paper.pdf), and they provided a count of 21,552 images in Table 2. This is closer to the size of our corpus. Regarding the one "missing" image, it could potentially be a result of file corruption, but I can't confirm this with absolute certainty at the moment.

**Comment by zwhus** [comment]
Thanks, When evaluating other datasets, are the parameters used consistent with those for WebQA, such as k? I made some simple modifications for FashionIQ, but the metrics show a slight difference. If the parameters are consistent, I will carefully check the code.

**Comment by JUNJIE99** [comment]
> Thanks, When evaluating other datasets, are the parameters used consistent with those for WebQA, such as k? I made some simple modifications for FashionIQ, but the metrics show a slight difference. If the parameters are consistent, I will carefully check the code.

Yes, since we only calculate up to the top-100, setting `k` to 100 would be sufficient. 

In addition, you need to pay attention to modifying the arguments in the `model.encode_*` within the `index` and `search` functions. For example, for FashionIQ, the `corpus_type` in the` search` function should be changed to `mm_it` because its query is image-text data.

**Comment by zwhus** [comment]
Thank you, I was able to reproduce the results. I mistakenly compared the results of m3 with the base.

**Comment by JUNJIE99** [comment]
Great!
